library/objective-c/EnvoyBridgeUtility.h

```diff
@@ -83,12 +83,27 @@ static inline EnvoyHeaders *to_ios_headers(envoy_headers headers) {
     NSString *headerValue = [[NSString alloc] initWithBytes:header.value.bytes
                                                      length:header.value.length
                                                    encoding:NSUTF8StringEncoding];
+    // Ensure list is present in dictionary value
     NSMutableArray *headerValueList = headerDict[headerKey];
     if (headerValueList == nil) {
       headerValueList = [NSMutableArray new];
       headerDict[headerKey] = headerValueList;
     }
-    [headerValueList addObject:headerValue];
+
+    // By convention, these headers disregard the RFC and contain commas single values.
+    if (headerKey == @"set-cookie" ||
+        headerKey == @"www-authenticate" ||
+        headerKey == @"proxy-authenticate") {
+      [headerValueList addObject:headerValue];
+    } else {
+      // Add trimmed, comma-separated values as individual members of the list.
+      NSArray *newValueList = [headerValue componentsSeparatedByString:@","];
+      for (NSString *value in newValueList) {
+        NSString *trimmedValue =
+          [value stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
```

Comment:
Is this necessary? I assume that envoy doesn't add white spaces, in which case we may more closely preserve the "original" content by not trimming

Reply:
It's desirable because comma delimiting may take the form of `","` or `", "`, and both are common. Users will likely find themselves inadvertently failing to match simple string comparisons if leading or trailing spaces are left on that are an artifact of how the header was coalesced and encoded.

Moreover, the RFCs state that all header values are encoded with optional leading and trailing whitespace, implying one should not treat it as part of the value.

Reply:
Sounds good. This implementation seems to have warnings when building:

```
INFO: From Compiling library/objective-c/EnvoyEngineImpl.m:
In file included from library/objective-c/EnvoyEngineImpl.m:2:
./library/objective-c/EnvoyBridgeUtility.h:94:19: warning: direct comparison of a string literal has undefined behavior [-Wobjc-string-compare]
    if (headerKey == @"set-cookie" || headerKey == @"www-authenticate" ||
                  ^  ~~~~~~~~~~~~~
./library/objective-c/EnvoyBridgeUtility.h:94:19: note: use 'isEqual:' instead
    if (headerKey == @"set-cookie" || headerKey == @"www-authenticate" ||
                  ^~
        [          isEqual:       ]
./library/objective-c/EnvoyBridgeUtility.h:94:49: warning: direct comparison of a string literal has undefined behavior [-Wobjc-string-compare]
    if (headerKey == @"set-cookie" || headerKey == @"www-authenticate" ||
                                                ^  ~~~~~~~~~~~~~~~~~~~
./library/objective-c/EnvoyBridgeUtility.h:94:49: note: use 'isEqual:' instead
    if (headerKey == @"set-cookie" || headerKey == @"www-authenticate" ||
                                                ^~
                                      [          isEqual:             ]
./library/objective-c/EnvoyBridgeUtility.h:95:19: warning: direct comparison of a string literal has undefined behavior [-Wobjc-string-compare]
        headerKey == @"proxy-authenticate") {
                  ^  ~~~~~~~~~~~~~~~~~~~~~
./library/objective-c/EnvoyBridgeUtility.h:95:19: note: use 'isEqual:' instead
        headerKey == @"proxy-authenticate") {
                  ^~
        [          isEqual:               ]
```

Reply:
Switched to `caseInsensitiveCompare`